get_primary_services/device_code.js

```diff
@@ -0,0 +1,72 @@
+/**
+ *  Copyright 2021 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+const data = new Uint32Array([3214]);
+let isConnected = false;
+let debug = '?';
+
+function updateScreen() {
+  if (typeof g === 'undefined') {
+    // Device has no graphics device (i.e. screen).
+    return;
+  }
+  g.clear();
+  g.setFontBitmap();
+  g.drawString('Current Notify Value:');
+  g.drawString('Connected: ' + isConnected, 0, g.getHeight() - 10);
+  let msg = 'Dbg: ' + debug;
+  g.drawString(msg, g.getWidth() - g.stringWidth(msg), g.getHeight() - 10);
+  g.setFontVector(40);
+  let val = data[0];
+  g.drawString(val, (g.getWidth() - g.stringWidth(val)) / 2, 12);
+
+  g.flip();
+}
+
+function onInit() {
+  // Put into a known state.
+  digitalWrite(LED, isConnected);
+
+  NRF.setServices({
+    '3f2b9742-7dce-11eb-9439-0242ac130002': {
```

Comment:
Since this is a test for getPrimaryServices() can you add more services or characteristics so we exercise a wider variety of properties?

Reply:
Good point - I should have been. Just pushed up a change that creates two services, each with two or more characteristics. The change is marked "WIP" because I cannot get the second service (6fc096cc-803b-11eb-9439-0242ac130002) to be displayed in the test. It does show up in chrome://bluetooth-internal (and Nordic connect). I don't see what I'm doing wrong - do you?

Reply:
You need to include that service in optionalServices.

Reply:
[sign] of course.